src/network-layer/default/RelayDefaultNetworkLayer.js

```diff
@@ -89,7 +89,8 @@ class RelayDefaultNetworkLayer {
         }
       }).catch(
         error => request.reject(error)
-      )
+      ),
+      request
```

Comment:
this doesn't seem to wait for `this._sendQuery(request)`, since it maps the `requests` back to themselves. what exact problem is this solving?